src/query/anonymization.c

```diff
@@ -270,6 +270,28 @@ static List *gather_aid_refs(Query *query, List *relations)
   return aid_refs;
 }
 
+static void reject_aid_grouping(Query *query, List *aid_refs)
+{
+  ListCell *cell;
+  List *grouping_exprs = get_sortgrouplist_exprs(query->groupClause, query->targetList);
+  foreach (cell, grouping_exprs)
+  {
+    Node *group_expr = (Node *)lfirst(cell);
+    if (IsA(group_expr, Var))
+    {
+      Var *var = (Var *)group_expr;
+
+      ListCell *aid_ref_cell;
+      foreach (aid_ref_cell, aid_refs)
+      {
+        AidRef *aid_ref = (AidRef *)lfirst(aid_ref_cell);
+        if (aid_ref->aid_attnum == var->varattno)
+          FAILWITH_LOCATION(var->location, "Selecting AID without generalization cannot yield any results - rejecting.");
```

Comment:
I picked the word "Selecting" here, but the problem lies actually in grouping/bucket expression. However, I'm not sure mentioning that is the most useful error message in this context. Anyone has ideas/thoughts on this?